Sources/SecureXPC/XPC Coders/XPCDecoder/XPCKeyedDecodingContainer.swift

```diff
@@ -161,4 +161,20 @@ internal class XPCKeyedDecodingContainer<K: CodingKey>: KeyedDecodingContainerPr
 	func superDecoder(forKey key: K) throws -> Decoder {
 		return XPCDecoderImpl(value: try value(forKey: key), codingPath: self.codingPath + [key])
 	}
+    
+    // MARK: XPC specific decoding
+    
+    func decodeEndpoint(forKey key: K) throws -> xpc_endpoint_t {
+        let value = try value(forKey: key)
+        let valueType = xpc_get_type(value)
+        if valueType != XPC_TYPE_ENDPOINT {
+            let message = "Expected type: \(XPC_TYPE_ENDPOINT.description)\nActual type: \(valueType.description)"
+            let context = DecodingError.Context(codingPath: self.codingPath,
+                                                debugDescription: message,
+                                                underlyingError: nil)
+            throw DecodingError.typeMismatch(xpc_object_t.self, context)
+        }
+        
+        return value
+    }
```

Comment:
Can't this just reuse `decode(key:xpcType:transform)` like most of the other methods areound here?

Something like:

```suggestion
    func decodeEndpoint(forKey key: K) throws -> xpc_endpoint_t {
        return try decode(key: key, xpcType: XPC_TYPE_ENDPOINT, transform: \.self)
    }
```

Reply:
Huh, yeah not sure why I didn't do that. However, I've never seen the `\.self` syntax before and it doesn't compile. This works as expected:
```swift
func decodeEndpoint(forKey key: K) throws -> xpc_endpoint_t {
    return try decode(key: key, xpcType: XPC_TYPE_ENDPOINT, transform: {$0})
}
```

Reply:
Huh, weird, I thought that was implemented already: https://github.com/apple/swift-evolution/blob/main/proposals/0227-identity-keypath.md